packages/adapters/storage/src/postgres.rs

```diff
@@ -479,47 +479,16 @@ impl Postgres {
         Ok(response)
     }
 
-    pub(crate) async fn total_unbundled_blocks(&self, starting_height: u32) -> Result<u64> {
-        let count = sqlx::query!(
-            r#"SELECT COUNT(*)
-                FROM fuel_blocks fb 
-                WHERE fb.height >= $1
-                AND NOT EXISTS (
-                    SELECT 1 FROM bundles b 
-                    WHERE fb.height BETWEEN b.start_height AND b.end_height
-                    AND b.end_height >= $1
-                )"#,
-            i64::from(starting_height)
-        )
-        .fetch_one(&self.connection_pool)
-        .await?
-        .count
-        .unwrap_or_default();
-
-        let count = u64::try_from(count)
-            .map_err(|_| crate::error::Error::Conversion("invalid block count".to_string()))?;
-
-        Ok(count)
-    }
-
     pub(crate) async fn _lowest_unbundled_blocks(
         &self,
         starting_height: u32,
-        max_cumulative_bytes: u32,
+        target_cumulative_bytes: u32,
+        block_buildup_threshold: u32,
     ) -> Result<Option<UnbundledBlocks>> {
-        // We're not using snapshot isolation here, so the count may change by the time
-        // we retrieve the blocks. If the pruner is configured aggressively to track the
-        // lookback window closely, the worst-case scenario is a temporary over- or
-        // underestimation of the block count. This discrepancy should resolve quickly,
-        // making the overhead of a snapshot-isolated transaction unnecessary.
-        let total_unbundled_blocks = self.total_unbundled_blocks(starting_height).await?;
-
-        if total_unbundled_blocks == 0 {
-            return Ok(None);
-        }
-
         let stream = self.stream_unbundled_blocks(starting_height);
-        let blocks = take_blocks_until_limit(stream, max_cumulative_bytes).await?;
+        let (blocks, buildup_detected) =
```

Comment:
Query that you send to the database also can limit the return result with `block_buildup_threshold`

Reply:
[Discussion here](https://github.com/FuelLabs/fuel-block-committer/pull/204/files#r2035336978)